rules/CodingStyle/Application/UseImportsRemover.php

```diff
@@ -34,14 +44,33 @@ public function removeImportsFromStmts(array $stmts, array $removedUses): array
 
     /**
      * @param string[] $removedUses
+     * @param AliasedObjectType[]|FullyQualifiedObjectType[] $useImportTypes
      */
-    private function removeUseFromUse(array $removedUses, Use_ $use): Use_
+    private function removeUseFromUse(array $removedUses, Use_ $use, array $useImportTypes): Use_
     {
+        // nothing to remove, as no replacement
+        if ($useImportTypes === []) {
+            return $use;
+        }
```

Comment:
on auto import enabled, when nothing to replace, the removing use statement should not happen, as it cause invalid replacement on rename process.